decred/decred/dcr/dcrdata.py

```diff
@@ -889,20 +888,21 @@ def sendOutputs(self, outputs, keysource, utxosource, feeRate=None):
           (dcrwallet/wallet/txauthor).AddAllInputScripts
 
         Args:
-            outputs (list(TxOut)): The transaction outputs to send.
+            outputs list(TxOut): The transaction outputs to send.
             keysource func(str) -> PrivateKey: A function that returns the
                 private key for an address.
-            utxosource func(int, func(UTXO) -> bool) -> list(UTXO): A function
-                that takes an amount in atoms, and an optional filtering
-                function. utxosource returns a list of UTXOs that sum to >= the
-                amount. If the filtering function is provided, UTXOs for which
-                the  function return a falsey value will not be included in the
-                returned UTXO list.
+            utxosource func(int, func(UTXO) -> bool) -> (list(UTXO), bool):
+                A function that takes an amount in atoms, and an optional
+                filtering function. utxosource returns a list of UTXOs that sum
+                to >= the amount, and a bool that states whether the funds are
+                sufficient to complete the transaction. If the filtering
+                function is provided, UTXOs for which the  function return a
+                falsey value will not be included in the returned UTXO list.
 
         Returns:
-            MsgTx: The sent transaction.
-            list(UTXO): The spent UTXOs.
-            list(UTXO): Length 1 array containing the new change UTXO.
+            newTx MsgTx: The sent transaction.
+            utxos list(UTXO): The spent UTXOs.
+            newUTXOs list(UTXO): Length 1 array containing the new change UTXO.
```

Comment:
It's a difficult tradeoff. I'd like to avoid the outer brackets for the `utxosource` description, but I also don't yet want to go and change the whole codebase to match. Not in this PR, at least. :slightly_smiling_face:

I also added names to the return values even though they are not part of Python syntax, as clarification and usage recommendation. I'm not yet ready to add these everywhere either.